src/gc/gc.cpp

```diff
@@ -34211,12 +34085,91 @@ HRESULT GCHeap::Initialize()
     uint32_t nhp_from_config = 0;
 
 #ifdef MULTIPLE_HEAPS
+    AffinitySet config_affinity_set;
+
+    // Get the affinity set configured by the user
+    uintptr_t heap_affinity_mask = GCConfig::GetGCHeapAffinitizeMask();
+    if (heap_affinity_mask != 0)
+    {
+        for (size_t i = 0; i < 8 * sizeof(uintptr_t); i++)
+        {
+            if (heap_affinity_mask & ((uintptr_t)1 << i))
+            {
+                config_affinity_set.Add(i);
+            }
+        }
+    }
+    else
+    {
+        GCConfigStringHolder cpu_index_ranges_holder(GCConfig::GetGCHeapAffinitizeRanges());
```

Comment:
>GCConfigStringHolder cpu_index_ranges_holder(GCConfig::GetGCHeapAffinitizeRanges()); [](start = 8, length = 84)

it would be really nice if all this config parsing code wasn't in gc.cpp. it seems better to do this in gcconfig.cpp... what do you think?

Reply:
I didn't like this location either, but wasn't sure what would be a better place. I like your suggestion.